.github/workflows/depends.yml

```diff
@@ -36,40 +36,62 @@ jobs:
         toolchain:
           - name: "RISCV 64bit"
             host: "riscv64-linux-gnu"
+            rust_host: "riscv64gc-unknown-linux-gnu"
             packages: "g++-riscv64-linux-gnu"
           - name: "ARM v7"
             host: "arm-linux-gnueabihf"
+            rust_host: "armv7-unknown-linux-gnueabihf"
             packages: "g++-arm-linux-gnueabihf"
           - name: "ARM v8"
             host: "aarch64-linux-gnu"
+            rust_host: "aarch64-unknown-linux-gnu"
             packages: "g++-aarch64-linux-gnu"
           - name: "i686 Linux"
             host: "i686-pc-linux-gnu"
+            rust_host: "i686-unknown-linux-gnu"
             packages: "g++-multilib"
           - name: "Win64"
             host: "x86_64-w64-mingw32"
+            rust_host: "x86_64-pc-windows-gnu"
             packages: "g++-mingw-w64-x86-64"
           - name: "x86_64 Linux"
             host: "x86_64-unknown-linux-gnu"
+            rust_host: "x86_64-unknown-linux-gnu"
           - name: "Cross-Mac x86_64"
             host: "x86_64-apple-darwin"
+            rust_host: "x86_64-apple-darwin"
             packages: "clang"
           - name: "Cross-Mac aarch64"
             host: "arm64-apple-darwin"
+            rust_host: "aarch64-apple-darwin"
             packages: "clang"
           - name: "x86_64 Freebsd"
             host: "x86_64-unknown-freebsd"
+            rust_host: "x86_64-unknown-freebsd"
             packages: "clang"
           - name: "ARMv7 Android"
             host: "arm-linux-android"
+            rust_host: "armv7-linux-androideabi"
           - name: "ARMv8 Android"
             host: "aarch64-linux-android"
+            rust_host: "aarch64-linux-android"
     name: ${{ matrix.toolchain.name }}
     steps:
     - name: set apt conf
       run: ${{env.APT_SET_CONF}}
     - name: install dependencies
       run: apt update; apt -y install build-essential libtool cmake autotools-dev automake pkg-config python3 gperf bsdmainutils curl git ca-certificates unzip ccache ${{ matrix.toolchain.packages }}
+    - name: install rust
+      # We can't use the latest Rust due to LLVM 17 not working with old `ld`s (such as in Ubuntu 20.04) for RISC-V
+      # We could update ld (a pain), update Ubuntu (requires a large amount of changes), or downgrade Rust
+      # We can't use Rust 1.70 due to LLVM 16 requiring ld >= 2.40 when building for Windows
+      run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

Comment:
Installing a script sourced from a serbian TLD without any validation in the depends environment makes me nervous. You may want to force a manual installation with a known checksum, as described here: https://rust-lang.github.io/rustup/installation/other.html#manual-installation.

Reply:
To be clear, I think country-level TLDs increase the risk (irrespective of country), but the methodological concern applies to any domain.

Reply:
Rustup installs unsigned binaries that rolled off some CI somewhere. For development builds, I think that's fine. For release builds, that's not okay. The CI jobs here are not release builds and anyone who wishes to run the artifacts should take proper precaution by running them in an isolated environment.

Reply:
> The CI jobs here are not release builds and anyone who wishes to run the artifacts should take proper precaution by running them in an isolated environment.

I do not object to this policy, and I like the wording; perhaps you can post this exact sentence as a comment at the top of the file, so readers understand the workflow's aim.